src/components/SiteSwitcher.tsx

```diff
@@ -14,14 +14,15 @@ import { SiteKey } from "../types/site"
 export const SiteSwitcher = () => {
   const [currentSite, setCurrentSite] = useAtom(siteAtom)
 
-  const switchSiteMutation = useMutation({
-    mutationFn: switchSite,
-    mutationKey: ["login"],
+  const refetchSiteContentMutation = useMutation({
```

Comment:
Rename to reflect what it should do: re-fetch the products and categories whenever the site changes.